### ISSUE TITLE
Publications ETL

### ISSUE THREAD

**Issue by ran-codes** [Drexel-UHC/salurbal-fair-renovations#165]
ETL issue for https://github.com/Drexel-UHC/salurbal-project-dashboard/issues/368

note that this is a renovation issue to ingest https://github.com/Drexel-UHC/salurbal-project-dashboard/files/12469518/SALURBAL.publications_OA_08.2023.v3.xlsx. Future inputs will be within the admin portal's CRUD application. 

## Action Items

 - [x] import existing publications metadata (https://github.com/Drexel-UHC/salurbal-project-dashboard/files/12469518/SALURBAL.publications_OA_08.2023.v3.xlsx)
 - [x] explore API structure
 - [x] harvest API to get metadata

**Comment by ran-codes** [comment]
So there are many formats and content from the PubMed API as per https://www.ncbi.nlm.nih.gov/pmc/tools/get-metadata/. It seems the XML format is most comprehensive the XML structure of interest for us as per the [example](https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=pubmed&id=33248227) 

- PubmedArticleSet
   - PubmedArticle
      - MedlineCitation 
         - **PMID** 
         - Article 
            - Journal
               - JournalIssue > **PubDate**
               - **Title**
            - **ArticleTitle**
            - Abstract > **AbstractText**
            - AuthorList  
               - Author
                  - **LastName**
                  - **ForeName**
                  - AffiliationInfo > **Affiliation**
         - KeywordList > **Keyword** 
      - PubmedData
         - ArticleIdList
            - ArticleId IdType="pubmed"
            - ArticleId IdType="pmc"
            - ArticleId IdType="doi"
            - ArticleId IdType="pii"

**Comment by ran-codes** [comment]
The XML for one of the SALURBAL papers looks like: https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=pubmed&id=31666140

looks okay. lets harvest.

**Comment by ran-codes** [comment]
Closing as ETL with Pubmed's efetch API is done.

**Comment by ran-codes** [comment]
## authorInfo Pubmed (PM) vs PubmedCentral (PMC) API

These two queries handle authors differently. For example for 'Life expectancy and mortality in 363 cities of Latin America' PubMed will return authors + `the SALURBAL group` but PubmedCentral will return all 100+ authors associated with `the SARLUABL group`. So because of these edge cases lets stick with PubMed authorInfo metadata.

**Comment by usamabilal** [comment]
Yep, agree. That's an extreme case (we only have a few more like taht i think). I'd keep it as SALURBAL group, so focus on PMID. Plus some stuff may not be in PMC (editorials for example)

**Comment by ran-codes** [comment]
got it! will focus on PMID metadata then. 

Closing. fixed this edge case.
